src/shortcuts/Shortcut.jsx

```diff
@@ -171,6 +173,24 @@ class Shortcut extends Context {
     get isComplete() {
         return true;
     }
+

```

Comment:
I don't think Shortcut model class should be responsible for creating a completion component. Let's have a separate factory method that is called from CompletionPortal and provides the shortcut or just its metadata as an argument and returns the correct completion component

Reply:
Four important things need to happen w/r/t completionComponents: 
- the CompletionPortalPlugin needs to be able to see whether or not a given shortcut we're neighboring has a completion component; 
- our CompletionPortal needs a reference to the completionComponent it is responsible for rendering; 
- our plugin needs a reference to the DOM node of he completion component so we can use that objects event handlers in our plugin; 
- our plugin receives as a prop a method that FNE defines, that uses a reference to the shortcut we're neighboring so that it knows which shortcut to update when a value is selected. 

The way I see it there are two more obvious ways this can be done: 

1) The Plugin and The Portal both have access to the factory, and we store a reference to the shortcut at the FNE editor level. 
- The Plugin can use a static method on the factory to see if the neighboring shortcut (passed through props) has a completion component and use a props.method to store a reference to that shortcut at the FNE level; 
- our portal will use the factory and a props.reference to the currently-neighbored shortcut to determine which completion component to render, using a props.method to bind a reference to that component to the FNE `this` object; 
- the Plugin uses some props.reference to the completionComponent DOM node passed through FNE to get relevant event handlers; 
- when a value is selected, a FNE function that was passed from FNE -> Portal -> CompletionComponent that can change the data of that shortcut.

2) The Plugin only has access to the factory, and we store references to both the completion component and the shortcut. This approach should be identical except that instead of determing through the factory which completionComponent we should be rendering inside our portal, we just use something that our CompletionPortal receives as a prop and render that as our completionComponent. That prop is a reference to the completionComponent class that we store  at the FNE level. 

If @gregquinn2001  you can let me know which of these is preferable I can continue working on the relevant changes tomorrow AM.